packages/ui/src/ui-component/input/TagInput.jsx

```diff
@@ -0,0 +1,94 @@
+import PropTypes from 'prop-types'
+import { useState, useEffect } from 'react'
+import { Box, Typography, Chip, Autocomplete, TextField, CircularProgress } from '@mui/material'
+import chatflowsApi from '@/api/chatflows'
+
+const TagInput = ({ categories, onChange }) => {
+    const [inputValue, setInputValue] = useState('')
+    const [allTags, setAllTags] = useState([])
+    const [loading, setLoading] = useState(true)
+
+    useEffect(() => {
+        fetchAllTags()
+    }, [])
+
+    const fetchAllTags = async () => {
+        try {
+            setLoading(true)
+            const response = await chatflowsApi.getAllChatflows()
+            const chatflows = response.data
+            const uniqueTags = [...new Set(chatflows.flatMap((chatflow) => (chatflow.category ? chatflow.category.split(';') : [])))]
+            setAllTags(uniqueTags)
+        } catch (error) {
+            console.error('Failed to fetch tags:', error)
+        } finally {
+            setLoading(false)
+        }
+    }
+
+    const handleTagChange = (event, newValue) => {
+        console.log('Tag changed:', newValue)
+        onChange(newValue)
```

Comment:
Need to validate that onChange is a function before calling it.   It _should_ be passed in, but I recommend checking